### PR TITLE
fix(stylelint): add stylelint to build process

### PR DESCRIPTION
**Pull request by mindreeper2420**
add Stylelint to the build process, with warnings and errors displayed as necessary

BREAKING CHANGE: build, if stylesheets are not written to spec

https://github.com/fabric8-ui/fabric8-planner/issues/2147

replaces original PR: https://github.com/fabric8-ui/fabric8-planner/pull/2150

### CODE REVIEW

File: .stylelintignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Ignore guidelines for stylelint
+# These are to be written in the same format as gitignore
+# Any directories or files are automatically found by the stylelint process
+
```
Comment (SMahil) [inline .stylelintignore]: we need to ignore formulas.less

**Comment by joshuawilson** [comment]
ping me if you need help on this one

**Comment by SMahil** [comment]
@mindreeper2420 This PR fails locally also. Its failing on "copy-watch-all" task

File: gulpfile.js [hunk 2]
```diff
@@ -14,14 +15,19 @@ var gulp = require('gulp'),
...
   cp = require('child_process'),
-  exec = require('gulp-exec'),
+  del = require('del'),
+  exec = require('gulp-exec').exec,
+  KarmaServer = require('karma').Server,
+  lessCompiler = require('gulp-less'),
+  stylelint = require ('gulp-stylelint'),
+  cssmin = require ('gulp-cssmin'),
   ngc = require('gulp-ngc'),
-  changed = require('gulp-changed'),
```
Comment (jarifibrahim) [inline gulpfile.js]: @mindreeper2420 The gulp build is failing with `changed` not found.  `gulp-changed` is used at https://github.com/mindreeper2420/fabric8-planner/blob/1f4abc328cc4ec741d065d609cc0844ca43f74ab/gulpfile.js#L52

**Comment by mindreeper2420** [comment]
I'll be updating this in the next day or so. Please do not merge master into this PR - I will do so locally and push an update as necessary.

File: gulpfile.js (hunk 2 above)
Comment (mindreeper2420) [reply gulpfile.js]: fixed - reference re-added.

**Comment by mindreeper2420** [comment]
@SMahil the build fails as there are stylelint errors.

If any command includes `gulp stylelint`, then the build will fail, as **failAfterError** is set to **true**.

If you run a command that utilizes `gulp lint-less`, then that command will not fail, as **failAfterError** is set to **false**.

You can turn off the fail on error, but then the build will still run and pass even with style issues. As noted in the command, this PR breaks Planner until these are fixed.

**Comment by joshuawilson** [comment]
@mindreeper2420 can you change the PR to submit the linting but not fail and then a new PR with the failing turned on. This will allow them to turn it on if they want and make it an issue to address or not.

**Comment by mindreeper2420** [comment]
@joshuawilson updated to remove failure on build errors, changed all errors alerts to be warnings and added a combiner to allow configuration of showing all errors in one grouping, rather than by line/file. I've added comments in the gulpfile where **false** needs to be set to **true** to turn build failures back on.

**Comment by mindreeper2420** [comment]
@joshuawilson @SMahil I pushed another set of changes, fixing files that had to be fixed in order to comply with stylelint. The task `gulp build:library`, which is required for `gulp copy-watch-all` now runs and completes.

I had to add a unit of measurement to the unit-whitelist setting, as I don't know what its intention actually is (the unit is `*pf-custom-navbar-item-height`). This should be fixed and removed from the whitelist in the future.

**Comment by joshuawilson** [comment]
@michaelkleinhenz it is very sad that @mindreeper2420 has tried twice now to get linting into the planner and it has been ignored.

**Pull request closed without merging**